aip/general/0182.md

```diff
@@ -0,0 +1,58 @@
+---
+id: 182
+state: reviewing
+created: 2021-08-13
+placement:
+  category: compatibility
+  order: 30
+---
+
+# External software dependencies
+
+Some services have a particular type of dependency on external software: they
+allow users to create resources that run on or expose the external software in
+some way. For example:
+
+- A database admin service can allow users to create databases running on a
+  particular version of a particular database engine (for example, PostgreSQL
+  13.4).
+- A virtual machine service can allow users to create VMs running a particular
+  operating system (for example, Ubuntu 20.04).
+- An application or function platform service can allow users to write code
+  that runs against a particular version of a programming language (for
+  example, Node.js 16.6).
+
+Services that provide external software to users in this way will eventually
+need to address the fact that all of these types of software have release
+lifecycles, and the versions they currently expose will eventually reach
+end-of-life.
+
+## Guidance
+
+Services that expose external software dependencies **should** allow users to
+create resources using any LTS (long-term support) version of the supported
```

Comment:
Any currently-supported LTS version, presumably. (There's more detail below already, admittedly.)

(And I wonder whether it's worth noting that it's okay for there to be a bit of a lag time between an LTS release coming out, and it being available - it's desirable to make new LTSes available quickly, but there can be cases where that's hard to do. Maybe that's too fiddly to express though.)

Reply:
Added "currently-supported".

I agree with the principle behind lag time, but I think it is implied, and the directive is a "should" anyway.